web/src/js/components/Modal/OptionTypes.jsx

```diff
@@ -0,0 +1,48 @@
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { update as updateOptions } from '../../ducks/options'
+
+MenuToggle.propTypes = {
+    value: PropTypes.bool.isRequired,
+    onChange: PropTypes.func.isRequired,
+    children: PropTypes.node.isRequired,
+}
+
+export function MenuToggle({ value, onChange, children }) {
+    return (
+        <div className="menu-entry">
+            <label>
+                <input type="checkbox"
+                        checked={value}
+                        onChange={onChange}/>
+                {children}
+            </label>
+        </div>
+    )
+}
+
+OptionsToggle.propTypes = {
+    option: PropTypes.string.isRequired,
+    children: PropTypes.node.isRequired,
+}
+
+export function OptionsToggle({ option, children, options, updateOptions }) {
+    return (
+        <MenuToggle
+            value={ options[option].value }
+            onChange={() => {console.log(options[option]);
+                updateOptions({ [option]: !(options[option].value)}) }}
+        >
+            {children}
+        </MenuToggle>
+    )
+}
+
+OptionsToggle = connect(
+    state => ({
+        options: state.options,
+    }),
+    {
+        updateOptions,
+    }
+)(OptionsToggle)
```

Comment:
This still looks a bit unintuitive to me. If we want to separate data ("OptionsToggle") from view ("MenuToggle"), we should make the data part a [higher-order component](https://facebook.github.io/react/docs/higher-order-components.html), so that we can just do
```javascript
MenuToggle = connectOptions(PureMenuToggle);
```
instead of defining our own data component for each datatype (i.e. instead of "OptionsToggle").

`connectOptions` could look like this for example:
```javascript
// this is untested
function connectOptions(WrappedComponent) {
    function Wrapper({name, options, updateOptions, ...props}){
        let option = options[name],
            onChange = (val) => updateOptions({ [option]: val });
        return <WrappedComponent name={name} value={option.value} onChange={onChange} {...props} />;
    }
    Wrapper.displayName = /*...*/;
    return connect(
        state = ({
            options: state.options,
        }),
        {
            updateOptions,
        }
    )(Wrapper)
}
```

I had a few beers earlier, I hope that makes sense. Ping me or Clemens on Slack if it doesn't. 😄